app/models/preorder_information.rb

```diff
@@ -34,9 +36,17 @@ def infer_status
     elsif school_will_order_devices? && any_school_users? && !chromebook_information_complete?
       'school_contacted'
     elsif school_will_order_devices? && any_school_users? && chromebook_information_complete?
-      'school_ready'
+      if school.std_device_allocation&.has_devices_available_to_order?
```

Comment:
```suggestion
      if school.can_order_devices_right_now?
```
I introduced this method on #616 so you'd need a rebase to pick it up.